File: java/src/main/java/codingdojo/DailyForecasts.java
package codingdojo;

import java.time.Instant;
import java.time.ZoneId;
import java.time.ZoneOffset;
import java.time.format.DateTimeFormatter;
import java.util.List;
import java.util.Map;
import java.util.Objects;

public class DailyForecasts {
    private final Scorer scorer = new Scorer();



    public void printSalesForecasts() {
        var names = List.of( "Steve", "Julie", "Francis");
        var formatter = DateTimeFormatter.ISO_LOCAL_DATE.withZone(ZoneId.from(ZoneOffset.UTC));
        System.out.println("Forecast at: " + formatter.format(Instant.now()));

        for (String name: names) {
            if (Objects.equals(name, "Steve"))
            {
                Scorer.flavour = IceCream.Strawberry;
            }
            else
            {
                scorer.updateSelection();
            }
            int score = scorer.getSalesForecast();
            System.out.println(name + " score: " + score);
        }
    }


}

File: kotlin/src/main/kotlin/org/codingdojo/DailyForecasts.kt
package org.codingdojo

import java.time.Instant
import java.time.ZoneId
import java.time.ZoneOffset
import java.time.format.DateTimeFormatter

class DailyForecasts {

    private val scorer: Scorer = Scorer()

    fun printSalesForecasts() {
        val names = listOf("Steve", "Julie", "Francis")
        val formatter = DateTimeFormatter.ISO_LOCAL_DATE.withZone(ZoneId.from(ZoneOffset.UTC))
        println("Forecast at: " + formatter.format(Instant.now()))
        for (name in names) {
            if (name == "Steve") {
                Scorer.flavour = IceCream.Strawberry
            } else {
                scorer.updateSelection()
            }
            val score: Int = scorer.getSalesForecast()
            println("$name score: $score")
        }
    }
}
